ui/app/controllers/settings/tokens.js

```diff
@@ -4,13 +4,18 @@ const { Controller, inject, computed, getOwner } = Ember;
 
 export default Controller.extend({
   token: inject.service(),
+  store: inject.service(),
 
   secret: computed.reads('token.secret'),
 
   tokenIsValid: false,
   tokenIsInvalid: false,
   tokenRecord: null,
 
+  resetStore() {
+    this.get('store').unloadAll();
```

Comment:
dang that is a big 🔨 (but I get why it's necessary) - are models unique across tokens or is there also a risk of id collisions here?

Reply:
No risk of collisions. Also no risk of leaking anything the user didn't at least have prior access to. It's just a Do The Right Thing™ change. If the token doesn't give access then it shouldn't be shown.